osbrain/proxy.py

```diff
@@ -100,6 +100,30 @@ def _ready_or_timeout(self, time0, timeout):
             raise
         return True
 
+    def wait_for_running(self, timeout=3.):
+        """
+        Wait until the agent is running.
+
+        Parameters
+        ----------
+        timeout : float
+            If not negative, raise an error if the agent is not running
+            after that number of seconds.
+
+        Returns
+        -------
+        Proxy
+            The object itself.
+        """
+        time0 = time.time()
+        while not self.get_attr('running'):
+            if timeout >= 0 and time.time() - time0 > timeout:
+                raise TimeoutError('Timed out while waiting for the agent to\
+                        run')
```

Comment:
Are you sure this exception will be appropriately formatted? Try it to see it for yourself.

If the printed message is not correctly formatted I'd rather use something like:

```python
# Shorter message, as the "TimeoutError" already tells you it is a timeout.
raise TimeoutError('Agent is not running')
```

Or:

```python
# Line split differently
raise TimeoutError('Asdf foo '
                   'bar qwer')
```

Or even:

```python
# Variable for the error message (more useful if you were to format it with variables)
error = 'Error message'
raise TimeoutError(error)
```

Reply:
@ocaballeror Could be redundant as we are already using `TimeoutError`, but is fine for me. :blush: